### DISCUSSION TITLE
[Kton DAO Temp Check] Optimizing KtonDAO Voting Process and Adjusting Treasury Reserve Allocation

### DISCUSSION THREAD

**Discussion by hackfisher** [darwinia-network/darwinia#1573] [General]
### Background:
Currently, the KtonDAO voting process on Tally has an excessively long decision cycle, leading to inefficiencies in proposal execution. The existing parameters require a 30-day voting period, and a 3-day delay before execution after the proposal passes. This prolonged process hinders agile decision-making within the DAO.

Additionally, with the regular distribution of RING from the Darwinia network, the incentive mechanism of KtonDAO requires more sustainable fund management. Presently, 20% of newly issued RING is regularly allocated to the KtonDAOVault contract, with 100% of it used for Kton Staking incentives. However, the KtonDAO Treasury lacks an independent reserve, limiting its ability to support potential strategic expenditures in the future.

### Proposal Details:
1. Adjusting Tally Voting Process Parameters:

Propose to increase the time from proposal publication to voting start from 0 to 1 day.
Propose to shorten the voting period from 30 days to 6 days.
Propose to maintain a 1-day delay from proposal passage to execution (currently 3 days).

The reason we can propose shortening the time cycle is that Tally has a 'proposal notify' feature, which allows security personnel to receive notifications and conduct checks more quickly. Otherwise, regular meetings would be required for inspections. It is recommended that everyone subscribes to https://notify.tally.xyz/ .

2. Establishing a Reserve Mechanism for the KtonDAO Treasury:

Currently, 20% of newly issued RING is regularly allocated to the KtonDAOVault contract, with 100% directed to Kton Staking incentives. This proposal suggests adjusting the allocation ratio to 80% for Kton Staking incentives and redirecting the remaining 20% to the KtonDAO Treasury Timelock contract as a reserve.
The specific use of these reserve funds is a topic of debate within the forum. It is recommended that, after the approval of this proposal, further proposals should be introduced to determine the usage through a DAO-wide vote.

### Additional Notes:
This proposal does not address the ongoing discussion regarding the continuation of Snowswap RING-KTON LP farm incentives. The execution of this proposal is expected to take 30-40 days. While I believe that continuous or short-term LP farm incentives are not necessarily required, if the discussion persists, I would recommend seeking temporary donations from DCDAO or other sources to support the community incentive.

### Reference

a. https://github.com/darwinia-network/KtonDAO/blob/d05e05d56682651110a40384fa6d8fadc2516305/script/Deploy.s.sol#L52-L59
b. https://github.com/orgs/darwinia-network/discussions/1566
c. https://github.com/orgs/darwinia-network/discussions/1571
d. https://github.com/orgs/darwinia-network/discussions/1544

**Comment by SasoLithops** [comment]
Absolutely agree on time shortening, except on this:

_propose to reduce the time from proposal publication to voting start from 0 to 1 day._
From my experiences I would propose at least 3 days so discussion can be made when one has free time and proposal can still be changed before it goes up for voting if something is wrong or could be executed better. 

Also I think since only 6 days will be available to vote, this address letting everyone know to subscribe https://notify.tally.xyz/ and link to https://www.tally.xyz/gov/ktondao/proposals to actually participate plus also a guide how to get gKTON should be posted for everyone to see on KTON governance page.


As for 2. point I am trying to find more attractive solution then just give up on 20% of rewards. Working now on one that will not require any additional Ring to be allocated and has roots in https://github.com/orgs/darwinia-network/discussions/1566 . In the mean time I personally would not mind this 20% to be allocated to treasury just to take part of it to continue funding LP farming incentives. Just not sure how we will be able to execute it since snowswap will probably need all RING in one transaction(that was for last 1.4M). If not, we can provide it in weekly transactions. Maybe like you recommended borrow from DCDAO and return in weekly/monthly payments. 
**But all of this would of course need to be discussed with KTON stakers.**

**Comment by cryptohustler101** [comment]
I favor a 3-day delay for discussions and a 6-day voting period, giving up to 9 days total. This strikes a balance between fast-tracking decisions and allowing for healthy discussions. Great work, @hackfisher @SasoLithops!

**Comment by sonoferin2000** [comment]
Reducing the voting cycle to 15 days offers a good starting point for balancing efficiency and inclusivity, ensuring that KtonDAO can act more swiftly while still allowing all members sufficient time to engage with proposals. The current 30-day voting period is cumbersome and delays important decisions, hindering the DAO's ability to respond to opportunities or challenges in a timely manner.

While the proposal to reduce the cycle to 6 days aims to increase efficiency, it may not be accommodating enough for those with busy schedules who need more time to thoroughly review and understand the proposals. A 15-day period provides a more moderate reduction, enhancing the DAO's agility while giving members the necessary time to familiarize themselves with the details, consult with peers, and make informed decisions. Simply put, 6 days is not enough time for everyone to fully engage in the process.

Starting with a 15-day voting cycle allows us to find a balance between speed and thoroughness. After implementing this change, we can reassess whether it strikes the right balance or if further adjustments are needed. This measured approach ensures that the KtonDAO remains responsive without compromising the quality of its decision-making process.

**Comment by Laki389** [comment]
> 
> 
> Propose to reduce the time from proposal publication to voting start from 0 to 1 day.

Just checking if there is a typo here, was it meant:

Propose to reduce the time from proposal publication to voting start from 1 to 0 day.

?

**Reply by hackfisher** [reply]
Sorry, it should be :


Propose to increase the time from proposal publication to voting start from 0 to 1 day.

**Comment by 2xJDubs** [comment]
I like the idea of making votes be a week long.  As you note, 30 days does not lend itself to agile decision making.

I support the idea of holding back 20% of KTON allocated chain emissions for funding the KTONDAO treasury.  Further more I think Laki has suggested transferring the KTON currently held by the RINGDAO treasury to the KTONDAO Treasury.  Which I think is a great idea.

**Reply by Laki389** [reply]
https://github.com/orgs/darwinia-network/discussions/1577

**Comment by nblogist** [comment]
I agree with the time adjustments and I think it is important for more agile movements as well. 
I don't have an opinion on Establishing a Reserve Mechanism for the KtonDAO Treasury currently.

**Comment by hackfisher** [comment]
Proposal published: https://www.tally.xyz/gov/ktondao/proposal/33324258396956465695619193347273649954830635754118257877158743750782579753923

**Reply by SasoLithops** [reply]
To those who nominated to me, I will vote Abstain as I am not against proposal but feel that 1 day is not enough for proposal to be discussed and adjusted (if necesary) by community feedback. 
I expressed that in my initial comment. I think it was three of us who commented on adjusting this data.

**Comment by hackfisher** [comment]
Proposal executed:

https://crab-scan.darwinia.network/address/0x08837De0Ae21C270383D9F2de4DB03c7b1314632?tab=coin_balance_history

https://explorer.darwinia.network/address/0x08837De0Ae21C270383D9F2de4DB03c7b1314632?tab=coin_balance_history

KtonDAO's treasury is accumulating RING.

**Reply by 2xJDubs** [reply]
It looks like something went wrong as the DAO treasury received 2 transfers but the last one was 16 hours ago.  There should have been 2 more in that time.

**Reply by hackfisher** [reply]
> It looks like something went wrong as the DAO treasury received 2 transfers but the last one was 16 hours ago.  There should have been 2 more in that time.

You mean the darwinia one, looks like some data display error or lag in the blockscout because they did not sum up.

The balance seems to be correct.

**Reply by 2xJDubs** [reply]
Yeah, looking at the address on Subscan it shows the 20% every 6hrs.  The explorer.darwinia.network is not showing all the txns for some reason.
